app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java

```diff
@@ -97,15 +105,18 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        FirebaseAuth auth = FirebaseAuth.getInstance();
-        if (auth.getCurrentUser() != null) {
-            startActivity(SignedInActivity.createIntent(this));
-            finish();
-        }
-
         setContentView(R.layout.auth_ui_layout);
         ButterKnife.bind(this);
 
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        if (user != null) {
+            if (!user.isAnonymous()) {
+                startActivity(SignedInActivity.createIntent(this));
```

Comment:
Now that you've refactored your approach, could you update the sample to show how it would look?  Don't worry about doing a perfect job, I just want to see a usage example that compiles